src/mlpack/tests/main_tests/preprocess_binarize_test.cpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * @file preprocess_binarize_test.cpp
+ * @author Manish Kumar
+ *
+ * Test mlpackMain() of preprocess_binarize_main.cpp.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <string>
+
+#define BINDING_TYPE BINDING_TYPE_TEST
+static const std::string testName = "PreprocessBinarize";
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/methods/preprocess/preprocess_binarize_main.cpp>
+
+#include <boost/test/unit_test.hpp>
+#include "../test_tools.hpp"
+
+using namespace mlpack;
+
+// Utility function to set a parameter and mark it as passed,
+// using copy semantics for lvalues and move semantics for rvalues.
+template<typename T>
+void SetInputParam(const std::string& name, T&& value)
+{
+  CLI::GetParam<typename std::remove_reference<T>::type>(name)
+       = std::forward<T>(value);
+  CLI::SetPassed(name);
+}
+
+struct PreprocessBinarizeTestFixture
+{
+ public:
+  PreprocessBinarizeTestFixture()
+  {
+    // Cache in the options for this program.
+    CLI::RestoreSettings(testName);
+  }
+
+  ~PreprocessBinarizeTestFixture()
+  {
+    // Clear the settings.
+    CLI::ClearSettings();
+  }
+};
+
+BOOST_FIXTURE_TEST_SUITE(PreprocessBinarizeMainTest,
+                         PreprocessBinarizeTestFixture);
+
+/**
+ * Check that input and output have same dimensions.
+ */
+BOOST_AUTO_TEST_CASE(PreprocessBinarizeDimensionTest)
+{
+  // synthetic dataset.
+  arma::mat inputData = arma::randu<arma::mat>(2, 5);
+
+  // Store size of input dataset.
+  size_t input_size  = inputData.n_cols;
```

Comment:
Can you rename this `inputSize` so that it conforms with the mlpack style guide? (no underscores in variable names)

https://github.com/mlpack/mlpack/wiki/DesignGuidelines#StyleGuidelines